search/services/index/classic.py

```diff
@@ -0,0 +1,106 @@
+"""Translate classic API `Phrase` objects to Elasticsearch DSL."""
+
+from typing import Dict, Callable
+
+from elasticsearch_dsl import Q, Search
+from elasticsearch_dsl.query import QueryString
+
+from ...domain import ClassicAPIQuery, Phrase, Term, Field, Operator
+from .prepare import SEARCH_FIELDS
+
+
+def classic_search(search: Search, query: ClassicAPIQuery) -> Search:
+    """
+    Prepare a :class:`.Search` from a :class:`.ClassicAPIQuery`.
+
+    Parameters
+    ----------
+    search : :class:`.Search`
+        An Elasticsearch search in preparation.
+    query : :class:`.ClassicAPIQuery`
+        An query originating from the Classic API.
+
+    Returns
+    -------
+    :class:`.Search`
+        The passed ES search object, updated with specific query parameters
+        that implement the advanced query.
+
+    """
+    search = search.filter("term", is_current=True)
+
+    # Initialize query
+    if query.phrase:
+        dsl_query = _phrase_to_query(query.phrase)
+    else:
+        dsl_query = Q()
+
+    # filter id_list if necessary
+    if query.id_list:
+        id_query = Q()
+        for id in query.id_list:
+            id_query &= (Q('terms', paper_id=query.id_list) 
+                         | Q('terms', paper_id_v=query.id_list))
```

Comment:
I thought this wasn't supported in classic?

Reply:
Some examples in classic:
http://export.arxiv.org/api/query?id_list=1509.07175
http://export.arxiv.org/api/query?id_list=1509.07175v5
http://export.arxiv.org/api/query?id_list=1509.07175v3

The behavior is with a paper_id, always return the most current, with the others, return the version specified.

Reply:
I've corrected the behavior of this - return most recent unversioned ids, return specified version otherwise. If no ids are passed, always return most recent.